featurebyte/session/snowflake.py

```diff
@@ -315,6 +316,12 @@ def get_columns_schema_from_dataframe(dataframe: pd.DataFrame) -> list[tuple[str
             ):
                 # Consider the type as an ARRAY if all elements are None, or a list.
                 db_type = "ARRAY"
+            elif (
+                dataframe.shape[0] > 0
+                and dataframe[colname].apply(lambda x: x is None or isinstance(x, dict)).all()
+            ):
+                # Consider the type as an OBJECT if all elements are None, or a dict.
+                db_type = "OBJECT"
```

Comment:
🤔 Should we remove the condition `dataframe[colname].apply(lambda x: x is None`?

If 
* `dataframe.shape[0] > 0`
* `dataframe[colname].apply(lambda x: x is None` is `True`
line 314-315 will be true.  It `db_type` will be set to `ARRAY`.  It seems to me that the additional check on  `dataframe[colname].apply(lambda x: x is None` is redundant.

Reply:
Not really. I think this check will still be true if dataframe column contains dicts and None values, right? 
If all values are none - yes, it will be marked as ARRAY, but if some values are none and some are dicts, we still wanna marked it as OBJECT. 
If I remove the check - dict columns with missing values stop being detected as OBJECT. 

Maybe I miss something?

Reply:
You're right. I misread this part as i thought is is something like
```
dataframe[colname].apply(lambda x: x is None)).all() or dataframe[colname].apply(lambda x: isinstance(x, dict)).all()
```